### PR TITLE
chore: refactoring bag

### PR DESCRIPTION
**Pull request by JackShort**
bag refactor

### CODE REVIEW

File: src/nft/hooks/useBagTotalEthPrice.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { useStablecoinValue } from 'hooks/useStablecoinPrice'
+import tryParseCurrencyAmount from 'lib/utils/tryParseCurrencyAmount'
+import { BagItemStatus } from 'nft/types'
+import { ethNumberStandardFormatter } from 'nft/utils'
+import { useMemo } from 'react'
+
+import { useBag } from './useBag'
+
+export function useBagTotalEthPrice(): BigNumber {
+  const itemsInBag = useBag((state) => state.itemsInBag)
```
Comment (cbachmeier) [inline src/nft/hooks/useBagTotalEthPrice.ts:13]: q: Did you consider adding these two functions to the `useBag` hook?

File: src/nft/components/bag/BagFooter.tsx [hunk 2]
```diff
@@ -375,7 +378,7 @@ export const BagFooter = ({ totalEthPrice, fetchAssets, eventProperties }: BagFo
     handleClick,
     buttonColor,
   } = useMemo(() => {
-    let handleClick = fetchAssets
+    let handleClick: (() => void) | (() => Promise<void>) = fetchAssets
```
Comment (cbachmeier) [inline src/nft/components/bag/BagFooter.tsx:381]: you should just be able to do `() => void`

File: src/nft/hooks/useSubscribeTransactionState.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  const { setBagStatus, setLocked: setBagLocked } = useBag(
+    ({ setBagExpanded, setBagStatus, setLocked }) => ({
+      setBagExpanded,
+      setBagStatus,
+      setLocked,
+    }),
+    shallow
+  )
+
+  useEffect(() => {
```
Comment (cbachmeier) [inline src/nft/hooks/useSubscribeTransactionState.ts:21]: q: did you get a chance to look into removing useEffects and instead bagStatus is directly updated only from the hooks in useFetchAssets or from direct user interaction

File: src/nft/hooks/useFetchAssets.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        const fetchedPriceChangedAssets = updatedAssets
+          .filter((asset) => asset.updatedPriceInfo)
+          .sort(sortUpdatedAssets)
+        const fetchedUnavailableAssets = updatedAssets.filter((asset) => asset.isUnavailable)
+        const fetchedUnchangedAssets = updatedAssets.filter((asset) => !asset.updatedPriceInfo && !asset.isUnavailable)
+        const hasReviewedAssets = fetchedUnchangedAssets.length > 0
+        const hasAssetsInReview = fetchedPriceChangedAssets.length > 0
+        const hasUnavailableAssets = fetchedUnavailableAssets.length > 0
+        const hasAssets = hasReviewedAssets || hasAssetsInReview || hasUnavailableAssets
+        const shouldReview = hasAssetsInReview || hasUnavailableAssets
```
Comment (cbachmeier) [inline src/nft/hooks/useFetchAssets.ts:131]: could we write a util that takes in `updatedAssets` or even `routeData and `assetsToBuy` and returns these params?

**Review by cbachmeier — COMMENTED** [review]
Minor nits but overall looks good. Ready to approve when route feature flag is removed

File: src/nft/components/bag/BagFooter.tsx (hunk 2 above)
Comment (JackShort) [reply src/nft/components/bag/BagFooter.tsx:380]: tsserver does not like that

File: src/nft/hooks/useFetchAssets.ts (hunk 4 above)
Comment (JackShort) [reply src/nft/hooks/useFetchAssets.ts:131]: this is in non feature flag code

File: src/nft/hooks/useSubscribeTransactionState.ts (hunk 3 above)
Comment (JackShort) [reply src/nft/hooks/useSubscribeTransactionState.ts:21]: not yet that is going to take some refactoring of useSendTransaction which is a rabbit hole that I haven't dove into yet

**Review by cbachmeier — APPROVED** [review]
LGTM

**Pull request merged**